src/backends/gpu/multi_event_stream.hpp

```diff
@@ -58,18 +59,20 @@ class multi_event_stream_base {
         n_nonempty_stream_(1)
     {}
 
+    // The list of events must be sorted in ascending order of index,
+    // with events of equal index further time-sorted.
     template <typename Event>
-    void init(std::vector<Event>& staged) {
+    void init(const std::vector<Event>& staged) {
         using ::arb::event_time;
```

Comment:
Did you have a measure of how much time is spent in the `multi_event_stream::init()` method? Given that the cell group has the partition-by-index information already, there is the possibility of reducing the work this call does. If the time spent here is inconsequential, then of course it is not important.

Reply:
I had that information, it was non-trivial, but the main bottleneck there was the `stable_sort`. Once that was no longer needed, it becomes very small.

I agree that things could be further optimized, but it isn't really needed at the moment.